languagetool-core/src/main/java/org/languagetool/rules/spelling/morfologik/MorfologikSpellerRule.java

```diff
@@ -220,6 +222,15 @@ protected Translator getTranslator(GlobalConfig globalConfig) {
     return null;
   }
 
+  // always returns false to follow `initSpellers` contract
+  private boolean initSpellersWithLock() throws IOException {
+    if (speller1 != null) return false;
```

Comment:
Such double-checked locking is only guaranteed to work if the checked variable `speller1` is volatile. I'd remove the outer check: it seems to lead to less scary code.